### PR TITLE
Tailwind

### PR DESCRIPTION
**Pull request by iruzevic**
### Added

- New way of handling Tailwind classes for forms.

### Fixed

- Rating field is now disabled if the parameter is set to true.
- Input number is now able to use standard tab navigation.
- Rating field now supports tab navigation.

### Removed

- Any unnecessary server side renders for the forms.

### CODE REVIEW

**Review by vedranpetrac — APPROVED** [review]

**Review by kancijan — APPROVED** [review]

File: src/Blocks/custom/forms/components/forms-editor.js [hunk 1]
```diff
@@ -69,23 +61,21 @@ export const FormsEditor = ({
...
-					)
-				}
-			/>
+			<Placeholder
+				icon={icons.form}
+				label={<span className='es-font-weight-400'>{__('Eightshift Forms', 'eightshift-forms')}</span>}
+				className='es-rounded-3! es-mx-auto! es-font-weight-400 es-color-cool-gray-500! es-nested-color-current!'
+				isColumnLayout={true}
+			>
+				{sprintf(__('Form "%s" with type "%s" will be displayed here.', 'eightshift-forms'), formsFormPostIdRaw?.label, formsFormPostIdRaw?.metadata)}
```
Comment (piqusy) [inline src/Blocks/custom/forms/components/forms-editor.js:75]: Translator info missing

File: src/Blocks/custom/forms/components/forms-editor.js [hunk 2]
```diff
@@ -94,34 +84,23 @@ export const FormsEditor = ({
...
-									}
-								/>
-							</>
+							<Placeholder
+								key={index}
+								icon={icons.form}
+								label={<span className='es-font-weight-400'>{__('Eightshift Forms', 'eightshift-forms')}</span>}
+								className='es-rounded-3! es-mt-5! es-mx-auto! es-font-weight-400 es-color-cool-gray-500! es-nested-color-current!'
+							>
+								{sprintf(__('Form "%s" with type "%s" will be displayed here.', 'eightshift-forms'), item?.form?.label, item?.form?.metadata)}
```
Comment (piqusy) [inline src/Blocks/custom/forms/components/forms-editor.js:98]: Translator info missing

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

Left a few suggestions re translations

File: src/Blocks/custom/forms/components/forms-editor.js (hunk 1 above)
Comment (iruzevic) [reply src/Blocks/custom/forms/components/forms-editor.js:75]: this is JS file it is not req here

File: src/Blocks/custom/forms/components/forms-editor.js (hunk 2 above)
Comment (iruzevic) [reply src/Blocks/custom/forms/components/forms-editor.js:98]: this is JS file it is not req here

File: src/Blocks/components/checkbox/checkbox.php [hunk 3]
```diff
@@ -39,21 +39,24 @@
 $checkboxFieldAttrs = Helpers::checkAttr('checkboxFieldAttrs', $attributes, $manifest);
 $checkboxIcon = Helpers::checkAttr('checkboxIcon', $attributes, $manifest);
 $checkboxIsHidden = Helpers::checkAttr('checkboxIsHidden', $attributes, $manifest);
+$checkboxTwSelectorsData = Helpers::checkAttr('checkboxTwSelectorsData', $attributes, $manifest);
 
 if ($checkboxAsToggle) {
 	$componentClass = "{$componentClass}-toggle";
 }
 
+$twClasses = FormsHelper::getTwSelectors($checkboxTwSelectorsData, ['checkbox']);
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/checkbox/checkbox.php:48]:
```suggestion
$twClasses = FormsHelper::getTwSelectors($checkboxTwSelectorsData, ['checkbox']);
```
Can `getTwSelectors` accept a variable number of args, so it behaves like the other Tailwind functions from Libs?

File: src/Blocks/components/error/error.php [hunk 4]
```diff
@@ -6,23 +6,26 @@
...
+$errorTwSelectorsData = Helpers::checkAttr('errorTwSelectorsData', $attributes, $manifest);
+
+$twClasses = FormsHelper::getTwSelectors($errorTwSelectorsData, ['error']);
 
 $errorClass = Helpers::classnames([
-	FiltersOuputMock::getTwSelectors($componentClass, $attributes),
+	FormsHelper::getTwBase($twClasses, 'error', $componentClass),
 	Helpers::selector($selectorClass, $selectorClass, $componentClass),
-	Helpers::selector($additionalErrorClass, $additionalErrorClass),
+	Helpers::selector($additionalClass, $additionalClass),
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/error/error.php:28]:
```suggestion
	$additionalClass,
```
Please do a find-replace for this in the whole plugin. The `classnames` handle will handle empty/null values.

File: src/Blocks/custom/forms/components/forms-editor.js [hunk 5]
```diff
@@ -94,34 +84,23 @@ export const FormsEditor = ({
...
-											}
-										)
-									}
-								/>
-							</>
+							<Placeholder
+								key={index}
+								icon={icons.form}
+								label={<span className='es-font-weight-400'>{__('Eightshift Forms', 'eightshift-forms')}</span>}
+								className='es-rounded-3! es-mt-5! es-mx-auto! es-font-weight-400 es-color-cool-gray-500! es-nested-color-current!'
```
Comment (goranalkovic-infinum) [inline src/Blocks/custom/forms/components/forms-editor.js:96]: These should be replaced with TW at some point, but let's leave them as-is for now 😄

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]

File: src/Blocks/components/checkbox/checkbox.php (hunk 3 above)
Comment (iruzevic) [reply src/Blocks/components/checkbox/checkbox.php:48]: this it not so similar to the libs helpers as this helper will filter out top level keys that could be used in this component. The array allows this to use multiple top level key like in the field component.

File: src/Blocks/components/error/error.php (hunk 4 above)
Comment (iruzevic) [reply src/Blocks/components/error/error.php:28]: I will do this in the refactoring of the components with updates on the libs, I will skip this at this point.

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request closed without merging**